### DISCUSSION TITLE
Kral ve mahkumlar programı

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus.github.io#69] [Soru Cevap]
Eski zamanlarda acımasız bir kral, on mahkumu daire biçiminde sıralar ve içlerinden birini seçerek ona bir kılıç verir. Kılıç kimdeyse solundaki ilk mahkumu öldürecek ve ölenin solundaki ilk canlı mahkuma kılıcı verecektir. İşlem tek bir mahkum kalana kadar devam edecek ve bu mahkum serbest bırakılacaktır. Kılıcın ilk verildiği mahkuma 1’den başlayarak sola doğru takip eden sayılar verilirse, bu on kişi içinden sırasıyla 2,4,6,8,10,3,7,1,9 numaralı mahkumlar ölecek ve 5 numaralı mahkum kurtulacaktır. Aynı işlem ONBİN mahkumluk bir gruba uygulansa hangi mahkum kurtulur programlayınız.

**Comment by TahaGuzell** [comment]
```
def kalan_mahkum(n):
    mahkumlar = list(range(1, n + 1))
    
    indeks = 0
    
    while len(mahkumlar) > 1:
        del mahkumlar[indeks]
        indeks = (indeks + 1) % len(mahkumlar)
    return mahkumlar[0]

sonuc =kalan_mahkum(10000)
print("Hayatta kalan mahkum:", sonuc)
```

**Comment by sonsuzus** [comment]
```python
kisiler = range(1,10001)
while len(kisiler)>1:
    kisiler = [*kisiler[2:], kisiler[0]]
print(kisiler)
```
